controllers/chat/initChatFromPost.js

```diff
@@ -289,9 +289,13 @@ const initChatFromPost = async (req, res) => {
 
     const response = {
       ...updatedChannel,
+      channel: {
+        ...updatedChannel?.channel,
+        messages: messageHistory?.results
+      },
       better_channel_members: betterChannelMember,
       better_channel_members_object: betterChannelMemberObject,
-      messageHistories: messageHistory.results
+      messageHistories: messageHistory?.results
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `channel` property is being restructured and the `messages` property is being added to it. This could potentially break any client-side code that relies on the previous structure of the `channel` property. It's important to ensure that all parts of the application that use this response are updated accordingly.

Also, there seems to be a redundancy in the response object. The `messageHistories` property at line 298 and the `messages` property inside the `channel` object at line 294 both contain `messageHistory?.results`. If these two properties are intended to serve the same purpose, consider removing one to avoid redundancy and potential confusion.

```diff
     const response = {
       ...updatedChannel,
       channel: {
         ...updatedChannel?.channel,
-        messages: messageHistory?.results
       },
       better_channel_members: betterChannelMember,
       better_channel_members_object: betterChannelMemberObject,
-      messageHistories: messageHistory?.results
+      messageHistories: messageHistory?.results
     };
     await client.disconnectUser();
```
In this suggested change, I've removed the `messages` property from the `channel` object and kept the `messageHistories` property at the root level of the response object. This will maintain the original structure of the `channel` object and still provide the message history data in the response.


<!-- This is an auto-generated comment by OSS CodeRabbit -->